src/pages/Login/Login.js

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable no-unused-vars */
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import HintInput from '../Input/HintInput';
+import './Login.scss';
+
+export const RULES = {
+  id: {
+    pattern: value => value.length > 5,
+    message: '아이디 양식을 확인하세요. 아이디는 최소 5자 이상이여야 합니다.',
+  },
+  password: {
+    pattern: value => value.includes('!', '@', '#', '$', '%', '^', '&'),
+    message:
+      '비밀번호 양식을 확인하세요. 비밀번호는 !, @, #, $, %, ^, &중 하나를 반드시 포함해야 합니다.',
+  },
+};
+
+const INIT_STATE = {
+  id: '',
+  password: '',
+};
+
+function Login() {
+  const [loginData, setLoginData] = useState(INIT_STATE);
+
+  const navigate = useNavigate();
+
+  const goToMain = () => {
+    if (
+      loginData.id.length > 5 &&
+      loginData.password.includes('!', '@', '#', '$', '%', '^', '&')
```

Comment:
위에 RULE을 만들어놨으니 이렇게 하는건 어때요?
```suggestion
      RULES.pattern(loginData.id) &&
      RULES.pattern(loginData.password)
```

여기서 이걸 더 간결하게 할 수 있는 메소드가 object를 array로 만들고 every라는 메소드를 사용하면 쉽게 전체가 Valide되어있는지 알 수 있을 것 같습니다!!